sdks/python/apache_beam/testing/load_tests/group_by_key_test.py

```diff
@@ -166,18 +188,31 @@ def setUp(self):
       raise ValueError('One or more of parameters for collecting metrics '
                        'are empty.')
 
+  class UngroupAndReiterate(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+      key, value = element
+      for i in range(self.iterations):
+        for v in value:
+          if i == self.iterations - 1:
+            return (key, v)
+
   def testGroupByKey(self):
-    # pylint: disable=expression-not-assigned
-    (self.pipeline
-     | beam.io.Read(synthetic_pipeline.SyntheticSource(
-         self.parseTestPipelineOptions()))
-     | 'Measure time: Start' >> beam.ParDo(
-         MeasureTime(self.metrics_namespace))
-     | 'GroupByKey' >> beam.GroupByKey()
-     | 'Ungroup' >> beam.FlatMap(
-         lambda elm: [(elm[0], v) for v in elm[1]])
-     | 'Measure time: End' >> beam.ParDo(MeasureTime(self.metrics_namespace))
-    )
+    input = (self.pipeline
+             | beam.io.Read(synthetic_pipeline.SyntheticSource(
+                 self.parseTestPipelineOptions()))
+             | 'Measure time: Start' >> beam.ParDo(
+                 MeasureTime(self.metrics_namespace))
+            )
+
+    for branch in range(self.fanout):
+      # pylint: disable=expression-not-assigned
+      (input
+       | 'GroupByKey %i' % branch >> beam.GroupByKey()
+       | 'Ungroup %i' % branch >> beam.FlatMap(
```

Comment:
It should use `UngroupAndReiterate` here, right?

Reply:
You are right! I fixed this